### ISSUE TITLE
[feat/short] SpreadFilter doesn't work with futures

### ISSUE THREAD

**Issue by ntsd** [freqtrade/freqtrade#6474]
I think because `bid` and `ask` are not in `ticker` in this line https://github.com/freqtrade/freqtrade/blob/5c9dddb7f357aebaf8f2a756e4977dffd1bcf26c/freqtrade/plugins/pairlist/SpreadFilter.py#L46

Freqtrade Version: 5c9dddb7f357aebaf8f2a756e4977dffd1bcf26c

Branch:  feat/short

Error: `Removed {pair} from whitelist due to invalid ticker data`

configs: 

```json
"pairlists": [
        {
			"method": "VolumePairList",
			"number_assets": 40,
			"sort_key": "quoteVolume",
			"refresh_period": 1800
		},
		{"method": "AgeFilter", "min_days_listed": 2},
		{"method": "SpreadFilter", "max_spread_ratio": 0.01},
		{
			"method": "RangeStabilityFilter",
			"lookback_days": 3,
			"min_rate_of_change": 0.03,
			"refresh_period": 1800
		},
		{
			"method": "VolatilityFilter",
			"lookback_days": 3,
			"min_volatility": 0.02,
			"max_volatility": 0.75,
			"refresh_period": 43200
		},
    ]
```

Logs: 
```txt
freqtrade    | 2022-02-27 19:56:32,290 - SpreadFilter - INFO - Removed DYDX/USDT from whitelist due to invalid ticker data: {'symbol': 'DYDX/USDT', 'timestamp': 1645991780414, 'datetime': '2022-02-27T19:56:20.414Z', 'high': 5.527, 'low': 5.035, 'bid': None, 'bidVolume': None, 'ask': None, 'askVolume': None, 'vwap': 5.311, 'open': 5.421, 'close': 5.056, 'last': 5.056, 'previousClose': None, 'change': -0.365, 'percentage': -6.733, 'average': 5.2385, 'baseVolume': 17686983.1, 'quoteVolume': 93934445.13, 'info': {'symbol': 'DYDXUSDT', 'priceChange': '-0.365', 'priceChangePercent': '-6.733', 'weightedAvgPrice': '5.311', 'lastPrice': '5.056', 'lastQty': '197.7', 'openPrice': '5.421', 'highPrice': '5.527', 'lowPrice': '5.035', 'volume': '17686983.1', 'quoteVolume': '93934445.130', 'openTime': '1645905360000', 'closeTime': '1645991780414', 'firstId': '119726231', 'lastId': '120027158', 'count': '300928'}}
freqtrade    | 2022-02-27 19:56:32,291 - SpreadFilter - INFO - Removed KNC/USDT from whitelist due to invalid ticker data: {'symbol': 'KNC/USDT', 'timestamp': 1645991780588, 'datetime': '2022-02-27T19:56:20.588Z', 'high': 2.047, 'low': 1.773, 'bid': None, 'bidVolume': None, 'ask': None, 'askVolume': None, 'vwap': 1.93677, 'open': 1.849, 'close': 1.903, 'last': 1.903, 'previousClose': None, 'change': 0.054, 'percentage': 2.92, 'average': 1.876, 'baseVolume': 45270487.0, 'quoteVolume': 87678722.92, 'info': {'symbol': 'KNCUSDT', 'priceChange': '0.05400', 'priceChangePercent': '2.920', 'weightedAvgPrice': '1.93677', 'lastPrice': '1.90300', 'lastQty': '4', 'openPrice': '1.84900', 'highPrice': '2.04700', 'lowPrice': '1.77300', 'volume': '45270487', 'quoteVolume': '87678722.92000', 'openTime': '1645905360000', 'closeTime': '1645991780588', 'firstId': '73775725', 'lastId': '74053006', 'count': '277282'}}
```

**Comment by xmatthias** [comment]
what exchange are you using this with?
The logs also show this for only 2 pairs - which can be ok (sometimes inactive pairs) - is it repeated for all pairs?

**Comment by ntsd** [comment]
> what exchange are you using this with? The logs also show this for only 2 pairs - which can be ok (sometimes inactive pairs) - is it repeated for all pairs?

The logs are longer than this basically every pair got this error. I removed the `SpreadFilter` and it's back to work.

I will try to debug this when I'm free.

exchange config
```
"exchange": {
        "name": "binance",
        "key": "",
        "secret": "",
        "ccxt_config": {},
        "ccxt_async_config": {},
        "pair_whitelist": [],
        "pair_blacklist": [
            // Exchange
            "(BNB)/.*",
            // Leverage
            ".*(_PREMIUM|BEAR|BULL|DOWN|HALF|HEDGE|UP|[1235][SL])/.*",
            // Fiat
            "(AUD|BRZ|CAD|CHF|EUR|GBP|HKD|IDRT|JPY|NGN|RUB|SGD|TRY|UAH|USD|ZAR)/.*",
            // Stable
            "(BUSD|CUSDT|DAI|PAXG|SUSD|TUSD|USDC|USDP|USDT|VAI)/.*",
            // FAN
            "(ACM|AFA|ALA|ALL|APL|ASR|ATM|BAR|CAI|CITY|FOR|GAL|GOZ|IBFK|JUV|LAZIO|LEG|LOCK-1|NAVI|NMR|NOV|OG|PFL|PORTO|PSG|ROUSH|SANTOS|STV|TH|TRA|UCH|UFC|YBO)/.*",
            // Others
            "(BOBA|CTXC|HBAR|NMR|SLP|OMG|ONG|ARDR)/.*"
        ]
    }
```

**Comment by xmatthias** [comment]
it seems like fetch_tickers ain't returning ask/bid prices (at least not on binance).

this means, we'll need to resort to "one call per pair" - which is probably not feasible since the moment you got the last pair, spread for the first pair will have changed, (potentialy significantly).

so at least for the moment, i'd assume that spreadFilter is not available on **binance futures**.
This is different in other futures markets - where ask/bid is filled (just tried on okx).

**Comment by xmatthias** [comment]
Closed by #6550

**Comment by hippocritical** [comment]
Hey sorry for ressurecting this one, I just found it would be better to have it in the same issue and not a new one.


from whitelist due to invalid ticker data

docker-container used: Running Freqtrade docker-2023.3.dev-b710bdaf (from "now")
Is it possible to be fixed in gateio or should I just not use spreadfilter?

2023-03-06 19:45:27 - SpreadFilter - INFO - Removed CFX/USDT:USDT from whitelist due to invalid ticker data: {'symbol': 'CFX/USDT:USDT', 'timestamp': None, 'datetime': None, 'high': 0.20951, 'low': 0.19515, 'bid': None, 'bidVolume': None, 'ask': None, 'askVolume': None, 'vwap': 0.19754999613961305, 'open': None, 'close': 0.19755, 'last': 0.19755, 'previousClose': None, 'change': None, 'percentage': -3.61, 'average': None, 'baseVolume': 145322220.0, 'quoteVolume': 28708404.0, 'info': {'total_size': '11810841', 'volume_24h_quote': '28708404', 'volume_24h_settle': '28708404', 'high_24h': '0.20951', 'change_percentage': '-3.61', 'volume_24h_base': '145322220', 'last': '0.19755', 'index_price': '0.197738', 'mark_price': '0.19775', 'funding_rate_indicative': '0.0001', 'funding_rate': '0.0001', 'contract': 'CFX_USDT', 'low_24h': '0.19515', 'volume_24h': '14532222'}}
2023-03-06 19:45:27 - SpreadFilter - INFO - Removed APT/USDT:USDT from whitelist due to invalid ticker data: {'symbol': 'APT/USDT:USDT', 'timestamp': None, 'datetime': None, 'high': 11.503, 'low': 10.956, 'bid': None, 'bidVolume': None, 'ask': None, 'askVolume': None, 'vwap': 11.236005735209364, 'open': None, 'close': 11.236, 'last': 11.236, 'previousClose': None, 'change': None, 'percentage': 0.83, 'average': None, 'baseVolume': 1497417.0, 'quoteVolume': 16824986.0, 'info': {'total_size': '8529255', 'volume_24h_quote': '16824986', 'volume_24h_settle': '16824986', 'high_24h': '11.503', 'change_percentage': '0.83', 'volume_24h_base': '1497417', 'last': '11.236', 'index_price': '11.24523', 'mark_price': '11.245', 'funding_rate_indicative': '-0.00012', 'funding_rate': '-0.00012', 'contract': 'APT_USDT', 'low_24h': '10.956', 'volume_24h': '14974178'}}
2023-03-06 19:45:27 - SpreadFilter - INFO - Removed LTC/USDT:USDT from whitelist due to invalid ticker data: {'symbol': 'LTC/USDT:USDT', 'timestamp': None, 'datetime': None, 'high': 90.69, 'low': 86.52, 'bid': None, 'bidVolume': None, 'ask': None, 'askVolume': None, 'vwap': 87.8700484928067, 'open': None, 'close': 87.87, 'last': 87.87, 'previousClose': None, 'change': None, 'percentage': -2.42, 'average': None, 'baseVolume': 173634.0, 'quoteVolume': 15257228.0, 'info': {'total_size': '3883768', 'volume_24h_quote': '15257228', 'volume_24h_settle': '15257228', 'high_24h': '90.69', 'change_percentage': '-2.42', 'volume_24h_base': '173634', 'last': '87.87', 'index_price': '87.91', 'mark_price': '87.91', 'funding_rate_indicative': '0.000073', 'funding_rate': '0.000073', 'contract': 'LTC_USDT', 'low_24h': '86.52', 'volume_24h': '1736341'}}
